src/Playwright/Transport/Converters/ProtocolSerializer.cs

```diff
@@ -115,54 +112,66 @@ public object Serialize(object value)
                 return new { r = new { p = regex.ToString(), f = regex.Options.GetInlineFlags() } };
             }
 
-            if (value is IDictionary dictionary && dictionary.Keys.OfType<string>().Any())
+            if (value is ExpandoObject)
             {
-                _visited.Add(value);
+                var o = new List<object>();
+                id = ++visitorInfo.LastId;
+                visitorInfo.Visited.Add(value, id);
```

Comment:
Visited is a Dictionary (hash table), and you are using a complex (potentially recursive) value as a key. In Java that would stack overflow unless identity hash code and comparison are used, so you'll need something like https://stackoverflow.com/questions/8946790/how-to-use-an-objects-identity-as-key-for-dictionaryk-v for all your Dictionaries.